smartsim/_core/generation/commandgenerator.py

```diff
@@ -0,0 +1,172 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import typing as t
+from os import path
+
+
+def move_op(to_move: str, dest_file: str):
+    """Write a python script to moves a file
+
+    :param to_move: Path to a source file to be copied
+    :param dest_file: Path to a file to copy the contents from the source file into
+    :return: string of python code to perform operation that can be executed
+    """
+    return f"import shutil;shutil.move('{to_move}','{dest_file}')"
+
+
+def delete_op(to_delete: str):
+    """Write a python script that deletes a file when executed.
+
+    :param to_delete: Path to the file to be deleted
+    :return: string of python code to perform operation that can be executed
+    """
+
+    if path.isfile(to_delete):
+        return rf"""
+import os
+os.remove('{to_delete}')
+    """
+    else:
+        raise FileNotFoundError(f"{to_delete} is not a valid path")
+
+
+def copy_op(to_copy: str, entity_path: str):
+    """
+    Write a python script to copy the entity files and directories attached
+    to this entity into an entity directory
+
+    :param to_copy: Path to directory, or path to file to copy into an entity directory
+    :param entity_path: Path to a directory of an entity
+    :return: string of python code to perform operation that can be executed
+    """
+
+    if not path.exists(entity_path):
+        raise FileNotFoundError(f"{entity_path} is not a valid path")
+    elif not path.exists(to_copy):
+        raise FileNotFoundError(f"{to_copy} is not a valid path")
+
```

Comment:
weird spacing here in the if elif else block

Reply:
changed